leshan-core/src/main/java/org/eclipse/leshan/core/util/TimestampUtil.java

```diff
@@ -0,0 +1,35 @@
+package org.eclipse.leshan.core.util;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+public class TimestampUtil {
+
+    public static BigDecimal fromInstant(Instant timestamp) {
+        if (timestamp == null) {
+            return null;
+        }
+
+        BigDecimal decimalPoints = new BigDecimal(timestamp.getNano()).divide(new BigDecimal(1_000_000_000));
+        return BigDecimal.valueOf(timestamp.getEpochSecond()).add(decimalPoints);
+    }
+
+    public static BigDecimal fromMilliseconds(long millis) {
+        return BigDecimal.valueOf(millis).divide(BigDecimal.valueOf(1000));
+    }
+
+    public static BigDecimal fromNanoseconds(long nanos) {
+        return BigDecimal.valueOf(nanos).divide(BigDecimal.valueOf(1_000_000_000));
+    }
```

Comment:
If this is not used now (or after all modifications), I think we should remove this 2 methods ?